src/ivc/fold_relaxed_plonk_instance_chip.rs

```diff
@@ -0,0 +1,493 @@
+use std::{iter, num::NonZeroUsize};
+
+use ff::{Field, FromUniformBytes, PrimeField, PrimeFieldBits};
+use halo2_proofs::circuit::Value;
+use halo2curves::{Coordinates, CurveAffine};
+use num_traits::Num;
+
+use crate::{
+    constants::NUM_CHALLENGE_BITS,
+    gadgets::{
+        ecc::{AssignedPoint, EccChip},
+        nonnative::bn::{
+            big_uint::{self, BigUint},
+            big_uint_mul_mod_chip::{self, BigUintMulModChip, ModOperationResult},
+        },
+    },
+    main_gate::{AssignedBit, AssignedValue, MainGate, MainGateConfig, RegionCtx, WrapValue},
+    nifs::CrossTermCommits,
+    plonk::{PlonkInstance, RelaxedPlonkInstance},
+    poseidon::ROCircuitTrait,
+    util,
+};
+
+pub(crate) struct FoldRelaxedPlonkInstanceChip<C: CurveAffine>
+where
+    C::Base: PrimeFieldBits + FromUniformBytes<64>,
+{
+    W: C,
+    E: C,
+    u: C::ScalarExt,
+    X0: BigUint<C::ScalarExt>,
+    X1: BigUint<C::ScalarExt>,
+
+    limb_width: NonZeroUsize,
+    limbs_count: NonZeroUsize,
+}
+
+struct AssignedWitness<C: CurveAffine> {
+    public_params_commit: AssignedValue<C::Base>,
+    W: AssignedPoint<C>,
+    E: AssignedPoint<C>,
+    u: AssignedValue<C::Base>,
+    assigned_X0: Vec<AssignedValue<C::Base>>,
+    assigned_X1: Vec<AssignedValue<C::Base>>,
+
+    plonk_W_commitment: AssignedPoint<C>,
+    plonk_instance: Vec<Vec<AssignedValue<C::Base>>>,
+    cross_terms_commits: Vec<AssignedPoint<C>>,
+
+    r: Vec<AssignedBit<C::Base>>,
+    m_bn: Vec<AssignedValue<C::Base>>,
```

Comment:
can you remind me what is `m_bn`?  Can add some comments to explain `r, m_bn`

Reply:
If I remember correctly, mod from diff field